src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -184,18 +263,18 @@ public void Write<T>([AllowNull] T value, string dataTypeName)
             //if (dataTypeName!= p.DataTypeName)
             //    throw new InvalidOperationException($"Can't change {nameof(p.DataTypeName)} from {p.DataTypeName} to {dataTypeName}");
 
-            Write(value, p);
+            await Write(value, p, async);
```

Comment:
This method doesn't need to be async - it can just return the task from Write on the last line.